### PR TITLE
fix: enable webPaas subscription environment upgrade

### PR DESCRIPTION
**Pull request by varun257**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #DTRSD-42203
| License          | BSD 3-Clause

## Description

enable webPaas subscription environment upgrade

### CODE REVIEW

**Review by frenautvh — DISMISSED** [review]

**Review by radireddy — APPROVED** [review]

**Review by antleblanc — APPROVED** [review]

**Review by JeremyDec — APPROVED** [review]

File: packages/manager/modules/web-paas/src/components/addon/controller.js [hunk 1]
```diff
@@ -35,6 +38,9 @@ export default class {
       const stagingQuantity = this.project.getTotalEnvironment();
       return stagingQuantity * addonPrice * this.quantity;
     }
+    if (this.addonType === ADDON_TYPE.ENVIRONMENT) {
+      return addonPrice * (this.quantity - DEFAULT_ENVIRONMENT);
```
Comment (marie-j) [inline packages/manager/modules/web-paas/src/components/addon/controller.js:42]: I expect `DEFAULT_ENVIRONMENT` to be something like a `string` not a `number`

File: packages/manager/modules/web-paas/src/components/additional-option/constants.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
 export const ADDON_FAMILY_STAGING_ENVIRONMENT = 'staging_environment';
 export const STORAGE_MULTIPLE = 5;
-export const DEFAULT_ENVIRONMENT = 2;
+export const DEFAULT_ENVIRONMENT_COUNT = 2;
```
Comment (ganeshkumar1989) [inline packages/manager/modules/web-paas/src/components/additional-option/constants.js:3]: Can we reuse this from packages/manager/modules/web-paas/src/web-paas.constants.js?

**Review by radireddy — APPROVED** [review]

**Review by ganeshkumar1989 — DISMISSED** [review]

File: packages/manager/modules/web-paas/src/components/addon/translations/Messages_fr_FR.json [hunk 3]
```diff
@@ -20,5 +20,6 @@
   "web_paas_service_add_addon_success_staging_environment": "La modification de votre environnements de test a bien été prise en compte. Vous pouvez suivre l’évolution de sa livraison dans votre compte, rubrique <a href='{{orderURL}}'>Mes commandes</a>",
   "web_paas_service_add_addon_success_user_license": "La modification de votre licences utilisateur a bien été prise en compte. Vous pouvez suivre l’évolution de sa livraison dans votre compte, rubrique <a href='{{orderURL}}'>Mes commandes</a>",
   "web_paas_service_add_addon_fetching_price": "Calcul du prix",
-  "web_paas_service_add_addon_storage_info": "Attention, il n'est pas possible de réduire le stockage d'un projet une fois le stockage additionnel commandé."
+  "web_paas_service_add_addon_storage_info": "Attention, il n'est pas possible de réduire le stockage d'un projet une fois le stockage additionnel commandé.",
+  "web_paas_service_add_addon_environment_info": "Attention, il n'est pas possible de réduire le nombre d’environnements de test d'un projet une fois l’environnements additionnel commandé."
```
Comment (antleblanc) [inline packages/manager/modules/web-paas/src/components/addon/translations/Messages_fr_FR.json:24]: > (...) une fois l’environnement**s** ...

Looks like there is a typo. `l'*` is singular so I think you can safely remove the extra `s`.

```suggestion
  "web_paas_service_add_addon_environment_info": "Attention, il n'est pas possible de réduire le nombre d’environnements de test d'un projet une fois l’environnement additionnel commandé."
```

File: packages/manager/modules/web-paas/src/components/addon/translations/Messages_en_GB.json [hunk 4]
```diff
@@ -16,5 +16,6 @@
   "web_paas_service_add_addon_success_staging_environment": "Your test environments have been modified. You can track the progress of the delivery in your account, in the <a href='{{orderURL}}'>My orders</a> section",
   "web_paas_service_add_addon_success_user_license": "Your user licences have been modified. You can track the progress of the delivery in your account, in the <a href='{{orderURL}}'>My orders</a> section",
   "web_paas_service_add_addon_fetching_price": "Price calculation",
-  "web_paas_service_add_addon_storage_info": "Warning: it is not possible to reduce the storage of a project once the additional storage has been ordered."
+  "web_paas_service_add_addon_storage_info": "Warning: it is not possible to reduce the storage of a project once the additional storage has been ordered.",
+  "web_paas_service_add_addon_environment_info": "Warning: The number of test environments for a project cannot be reduced once the additional environments have been ordered."
```
Comment (antleblanc) [inline packages/manager/modules/web-paas/src/components/addon/translations/Messages_en_GB.json:20]: **question:** Please refer to my previous comment. Looks like there is a typo, isn't it?

```suggestion
  "web_paas_service_add_addon_environment_info": "Warning: The number of test environments for a project cannot be reduced once the additional environment have been ordered."
```

File: packages/manager/modules/web-paas/src/web-paas.constants.js [hunk 5]
```diff
@@ -55,7 +55,7 @@ export const LICENCES_VALUES = {
   MAX: 100,
 };
 export const STORAGE_MULTIPLE = 5;
-export const DEFAULT_ENVIRONMENT = 2;
+export const DEFAULT_ENVIRONMENT_COUNT = 2;
```
Comment (antleblanc) [inline packages/manager/modules/web-paas/src/web-paas.constants.js:58]: **question:** Any reason to not export it also as `default` ? (see below L60-73)

**Review by antleblanc — CHANGES_REQUESTED** [review]
Found some existing some typos.
Could you please make sure that there's no more typos before releasing it?
Thank  you!

**Comment by varun257** [comment]
@antleblanc have taken care of the changes requested

**Review by antleblanc — CHANGES_REQUESTED** [review]
Changes looks good to me. Only some recent changes around the content implies
to redo the translation process I think.

For instance, `Messages_fr_CA.json` is no more aligned with` Messages_fr_FR.json`.

**Comment by ovh-cds** [comment]
CDS Report translator-push#19011.0 ✘  
* 
  * push ✘

**Comment by varun257** [comment]
> Changes looks good to me. Only some recent changes around the content implies
> to redo the translation process I think.
> 
> For instance, `Messages_fr_CA.json` is no more aligned with` Messages_fr_FR.json`.

Hi @antleblanc , They are okay with the translation as is for now as they need to get this prodded as soon as possible and translation process might take more time. So we have a go ahead to prod this with FR fix. PU can raise an issue for this later.

**Comment by varun257** [comment]
QA done and approved by QA_TEAM, We can include this in the next release

**Comment by antleblanc** [comment]
Updating target branch from `master` to `develop`.

**Comment by antleblanc** [comment]
Target branch has been updated successfully. Now target `develop`.

**Review by antleblanc — APPROVED** [review]

**Pull request merged**